lib/backend/kubernetes/kubernetes.go

```diff
@@ -191,15 +193,23 @@ func NewSharedWithClient(restClient kubernetes.Interface) (*Backend, error) {
 }
 
 // NewWithConfig returns a new instance of Kubernetes Secret identity backend storage with the provided config.
-func NewWithConfig(conf Config) (*Backend, error) {
+// Ensures that the secret exists.
+func NewWithConfig(ctx context.Context, conf Config) (*Backend, error) {
 	if err := conf.Check(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	return &Backend{
+	b := &Backend{
 		Config: conf,
-		mu:     sync.Mutex{},
-	}, nil
+	}
+	if b.Exists(ctx) {
+		return b, nil
+	}
+	// NOTE: This will trigger an actual Create() since we don't have a resource version set.
+	// If we have an AlreadyExists error, discard it and move on as it can happen with rapid restarts/rollouts/agent name collisions.
+	if err := b.createSecret(ctx); err != nil && !trace.IsAlreadyExists(err) {
```

Comment:
I don't think this will ever return trace.IsAlreadyExists. It will return the kubernetes error and not our internal trace.AlreadyExists error